limit working now with binomial #10801

wrat opened the pull request:
As we know that binomial funtion can be reduced into to their factorial form
 for calculating the limit of funtion we are using gruntz algorithm. gruntz algorithm is not works with  binomial limits so we have to convert its factorial form.
 for example:
 while calculating `limit(16**k/(k*binomial(2*k,k)**2), k, oo)`
 we are getting NotImplemented Exception but after some changes we are now able to get the result as PI.
"

wrat committed:
"As we know that binomial funtion can be reduced into to their factorial form.
 for calculating the limit of funtion we are using gruntz algorithm. gruntz algorithm is not works with binomial limits so we have to convert its factorial form.
 for example:
 while calculating limit(16**k/(k*binomial(2*k,k)**2), k, oo)
 we are getting NotImplemented Exception but after some changes we are now able to get the result as PI.
 Refer to #10801 issue
"

wrat committed:
"As we know that binomial funtion can be reduced into to their factorial form.
 for calculating the limit of funtion we are using gruntz algorithm. gruntz algorithm is not works with binomial limits so we have to convert its factorial form.
 for example:
 while calculating limit(16**k/(k*binomial(2*k,k)**2), k, oo)
 we are getting NotImplemented Exception but after some changes we are now able to get the result as PI.
"

wrat committed:
Merge branch 'master' into 1729_branch

smichr commented on sympy/series/limits.py:
```diff
@@ -135,7 +136,12 @@ def doit(self, **hints):
         if not e.has(z):
             return e
-
+        #limit is not working with binomial,gruntz raise an exception
```
space after `#` and space around `=` below

wrat replied:
So it is ready  now to get merged ?

wrat replied:
what else should I add in my commit message??

smichr replied:
> ready now to get merged

No yet. Please add spaces after the `#` and format your commit message with a blank line after the title. Please see the reference/link that I referred to if you have questions.

smichr commented:
Please edit your commit message:

:eyeglasses: see requirements [here](https://github.com/sympy/sympy/wiki/Development-workflow#writing-commit-messages)

smichr commented:
Other than the comments I made, this seems like a good idea. Thanks for identifying this improvement.

wrat committed:
" Put some spaces here

          if not e.has(z):
              return e
-
+        #limit is not working with binomial,gruntz raise an exception

     and   e = e.rewrite(factorial)

   issue  #10801
"

wrat committed:
Merge branch '1729_branch' of github.com:wrat/sympy into 1729_branch
" space after # and space around = below in limits.py
  issue #10801
"
Conflicts:
	sympy/series/limits.py
	sympy/series/tests/test_limits.py

wrat committed:
"As we know that binomial funtion can be reduced into to their factorial form.for calculating the limit of funtion we are using gruntz algorithm. gruntz algorithm is not works with binomial limits so we have to convert its factorial form.
for example:
while calculating limit(16**k/(k*binomial(2*k,k)**2), k, oo)
we are getting NotImplemented Exception but after some changes we are now able to get the result as PI.
Refer to #10801 issue"

wrat committed:
"Remove trailing white space"

wrat commented:
@asmeurer ping!!

smichr commented on sympy/series/limits.py:
```diff
@@ -136,12 +136,13 @@ def doit(self, **hints):
         if not e.has(z):
             return e
-        #limit is not working with binomial,gruntz raise an exception
+
```
I didn't mean put line breaks after the `#` I meant to change `#foo` to `# foo` -- space after the `#`

a space after the `binomial,`, too

wrat replied:
ping @smichr

wrat committed:
"Add some spaces after # and binomial, "

wrat commented:
ping @asmeurer

leosartaj commented:
Closing in favor of #10883.
